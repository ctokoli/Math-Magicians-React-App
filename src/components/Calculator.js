import React, { useState } from 'react';
import calculate from '../logic/calculate';
import Values from '../util/keyboard';
import Buttons from '../util/button';

function Calculator() {
  const [result, setResult] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const buttonHandler = (btn) => {
    setResult(calculate(result, btn));
  };

  return (
    <div className="calculator">
      <div className="result"><p>{result.next || result.operation || result.total || '0'}</p></div>
      {Values.map((value) => (
        <Buttons
          classN={value.class}
          onclick={(e) => buttonHandler(e.target.textContent)}
          input={value.key}
          key={value.id}
        />
      ))}
    </div>
  );
}

export default Calculator;
