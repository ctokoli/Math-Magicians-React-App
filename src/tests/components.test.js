import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Calculator from '../components/Calculator';
import HomePage from '../components/Home';
import LayoutPage from '../components/Layout';
import DisplayQuotes from '../components/Quotes';

describe('Components Testing', () => {
  it('Should test for the calculator component', () => {
    const calculator = renderer.create(<Calculator />).toJSON();
    expect(calculator).toMatchSnapshot();
  });

  it('calculator component should render all the buttons ', () => {
    render(<Calculator />);
    const buttons = screen.getAllByRole('button');
    expect(buttons).toHaveLength(19);
  });

  it('home component should render text ', () => {
    render(<HomePage />);
    const heading = screen.getByText('This is the home page');
    expect(heading).toBeInTheDocument();
  });

  it('Should test for the HomePage component', () => {
    const home = renderer.create(<HomePage />).toJSON();
    expect(home).toMatchSnapshot();
  });

  it('Should test for the layout component', () => {
    const layout = renderer.create(
      <MemoryRouter>
        <LayoutPage />
      </MemoryRouter>,
    ).toJSON();
    expect(layout).toMatchSnapshot();
  });

  it('Should test for the Quotes component', () => {
    const home = renderer.create(<DisplayQuotes />).toJSON();
    expect(home).toMatchSnapshot();
  });
});
