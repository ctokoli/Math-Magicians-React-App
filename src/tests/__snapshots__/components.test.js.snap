// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Testing Should test for the calculator component 1`] = `
<div
  className="calculator"
>
  <div
    className="result"
  >
    <p>
      0
    </p>
  </div>
  <button
    className="AC"
    onClick={[Function]}
    type="button"
  >
    AC
  </button>
  <button
    className="plusnsub"
    onClick={[Function]}
    type="button"
  >
    +/-
  </button>
  <button
    className="percentage"
    onClick={[Function]}
    type="button"
  >
    %
  </button>
  <button
    className="division orange"
    onClick={[Function]}
    type="button"
  >
    รท
  </button>
  <button
    className="seven"
    onClick={[Function]}
    type="button"
  >
    7
  </button>
  <button
    className="eight"
    onClick={[Function]}
    type="button"
  >
    8
  </button>
  <button
    className="nine"
    onClick={[Function]}
    type="button"
  >
    9
  </button>
  <button
    className="multi orange"
    onClick={[Function]}
    type="button"
  >
    x
  </button>
  <button
    className="four"
    onClick={[Function]}
    type="button"
  >
    4
  </button>
  <button
    className="five"
    onClick={[Function]}
    type="button"
  >
    5
  </button>
  <button
    className="six"
    onClick={[Function]}
    type="button"
  >
    6
  </button>
  <button
    className="substract orange"
    onClick={[Function]}
    type="button"
  >
    -
  </button>
  <button
    className="one"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="two"
    onClick={[Function]}
    type="button"
  >
    2
  </button>
  <button
    className="three"
    onClick={[Function]}
    type="button"
  >
    3
  </button>
  <button
    className="plus orange"
    onClick={[Function]}
    type="button"
  >
    +
  </button>
  <button
    className="zeroo"
    onClick={[Function]}
    type="button"
  >
    0
  </button>
  <button
    className="dott"
    onClick={[Function]}
    type="button"
  >
    .
  </button>
  <button
    className="equal orange"
    onClick={[Function]}
    type="button"
  >
    =
  </button>
</div>
`;
